### PR TITLE
net-ftp/tftp-hpa: bump EAPI and split the package

### PR DESCRIPTION
**Pull request by oz123**
Now a user can install either the tftp client or the tftp server
or both. This is controlled by use flags. By default both are installed,
so this keeps backwards compatibility.

Package-Manager: Portage-2.3.103, Repoman-2.3.23
Signed-off-by: Oz Tiram <oz.tiram@gmail.com>

### CODE REVIEW

**Comment by oz123** [comment]
@thesamesam as requested, here is a PR to split tftp-hpa a-la debian's split package, so one can install only the client or the server or both as needed.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+src_install() {
+	dodoc README* CHANGES tftpd/sample.rules
+	emake INSTALLROOT="${D}" -C lib install
+	emake INSTALLROOT="${D}" -C common install
+
+	if use client; then
+		emake INSTALLROOT="${D}" -C tftp install
+	fi
+	if use daemon; then
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:64]: I think this should be `if use server`.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+		emake INSTALLROOT="${D}" install
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:66]: Doesn't this end up installing the client as well? Or does the toplevel Makefile not recurse into the tftp directory?

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+src_compile() {
+	emake version.h
+	emake -C lib
+	emake -C common
+	if use client; then
+		emake -C tftp
+	fi
+	if use daemon; then
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:51]: I think this should be `if use server`.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:4]: Bump it to EAPI 7 please.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 1 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:64]: Oh, my bad ... I change the use flag, but the ebuild still uses my older terminology ...

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 2 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:66]: Yes, that's actually correct. The makefile from upstream provides a veriable to control this. I updated the ebuild and verified that it works.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 3 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:51]: Yes.  I updated this.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 4 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild]: Done.Thanks for that review.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit systemd eutils toolchain-funcs
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:6]: Drop eutils.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+LICENSE="BSD-4"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~ppc-macos"
+IUSE="ipv6 readline selinux tcpd +client +server"
+
+CDEPEND="
+	readline? ( sys-libs/readline:0= )
+	tcpd? ( sys-apps/tcp-wrappers )
+	!net-ftp/atftp"
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:20]: Should probably move this to RDEPEND only.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	fi
+	if use server; then
+		emake -C tftpd
+	fi
+}
+
+src_install() {
+	dodoc README* CHANGES tftpd/sample.rules
+	emake INSTALLROOT="${D}" -C lib install
+	emake INSTALLROOT="${D}" -C common install
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:59]: Based on reading the Makefiles, these two lines do nothing.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+	if use client; then
+		emake INSTALLROOT="${D}" SUB="lib common tftp" tftp install
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:62]: This does not seem right: you are passing 2 targets to make: "tftp" and "install".

Also, I don't think the SUB variable is necessary. Does this work?
```
emake INSTALLROOT="${D}" -C tftp install
```

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+	fi
+	if use server; then
+
+		emake INSTALLROOT="${D}" SUB="lib common tftpd" tftpd install
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:66]: Same issue as the client install.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+		# iputils installs this
+		rm "${ED}"/usr/share/man/man8/tftpd.8 || die
```
Comment (floppym) [inline net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:68]: This seems a bit odd. Does iputils not install a conflicting binary as well?

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 8 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:62]: Yes, that works too.

Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:62]: Your suggestion works too.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 10 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:68]: Seems like a left over from a distant past. I don't why this was added (The comment exists in the original commit from Sat Aug 8 13:49:04 2015 -0700). I guess this comment be removed. 

```
equery f iputils
 * Searching for iputils ...
 * Contents of net-misc/iputils-20190709-r1:
/bin
/bin/arping
/bin/ping
/bin/ping4 -> ping
/bin/ping6 -> ping
/usr
/usr/bin
/usr/share
/usr/share/doc
/usr/share/doc/iputils-20190709-r1
/usr/share/doc/iputils-20190709-r1/README.md.bz2
/usr/share/locale
/usr/share/locale/ja
/usr/share/locale/ja/LC_MESSAGES
/usr/share/locale/ja/LC_MESSAGES/iputils.mo
/usr/share/man
/usr/share/man/man8
/usr/share/man/man8/arping.8.bz2
/usr/share/man/man8/ping.8.bz2
```

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 7 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild:59]: Yes. It seems that these are oboslete.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 5 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild]: Done.

File: net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild (hunk 6 above)
Comment (oz123) [reply net-ftp/tftp-hpa/tftp-hpa-5.2-r2.ebuild]: Done.

**Pull request closed without merging**